### PR TITLE
Add lookgroup and collection classes.

### PR DESCRIPTION
**Pull request by bernardkwok**
Update: LOOKDEVX-144
- Add in RtLookGroup, and RtCollection.
- Small updates to RtLook so that RtMaterialAssign are connection relationships.
- Add in unit test.

### CODE REVIEW

File: source/MaterialXRuntime/RtCollection.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
+//
+// TM & (c) 2019 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
```
Comment (feldstj) [inline source/MaterialXRuntime/RtCollection.cpp]: minor: 2020?

File: source/MaterialXRuntime/RtLook.cpp [hunk 2]
```diff
@@ -16,11 +16,58 @@ namespace
...
+
+RtAttribute RtLookGroup::getActiveLook() const
+{
+    return prim()->getAttribute(ACTIVELOOK)->hnd();
+}
+
+void RtLookGroup::addLook(const RtObject& look)
+{
+    RtRelationship rel = getLooks();
+    rel.removeTarget(look);
```
Comment (feldstj) [inline source/MaterialXRuntime/RtLook.cpp]: Aren't your addLook and removeLook functions backwards?

File: source/MaterialXRuntime/RtCollection.h [hunk 3]
```diff
@@ -0,0 +1,36 @@
+//
+// TM & (c) 2019 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
```
Comment (feldstj) [inline source/MaterialXRuntime/RtCollection.h]: minor: 2020?

File: source/MaterialXRuntime/RtCollection.h [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+/// Classes related to collections of geometry identifiers
+
+#include <MaterialXRuntime/RtSchema.h>
+
+namespace MaterialX
+{
+
+/// @class RtCollection
+/// Schema for collection prims.
+class RtCollection: public RtTypedSchema
```
Comment (feldstj) [inline source/MaterialXRuntime/RtCollection.h]: Don't you need setters as well?

**Comment by feldstj** [comment]
You should probably add a test case as well.

File: source/MaterialXRuntime/RtLook.cpp (hunk 2 above)
Comment (bernardkwok) [reply source/MaterialXRuntime/RtLook.cpp]: Yes :)

File: source/MaterialXRuntime/RtCollection.h (hunk 4 above)
Comment (bernardkwok) [reply source/MaterialXRuntime/RtCollection.h]: Added now.

File: source/MaterialXRuntime/RtLook.cpp [hunk 5]
```diff
@@ -16,11 +16,59 @@ namespace
...
+DEFINE_TYPED_SCHEMA(RtLookGroup, "lookgroup");
+
+RtPrim RtLookGroup::createPrim(const RtToken& typeName, const RtToken& name, RtPrim parent)
+{
+    if (typeName != _typeName)
+    {
+        throw ExceptionRuntimeError("Type names mismatch when creating prim '" + name.str() + "'");
+    }
+
+    const RtToken primName = name == EMPTY_TOKEN ? LOOK1 : name;
```
Comment (niklasharrysson) [inline source/MaterialXRuntime/RtLook.cpp]: I think LOOK1 should be LOOKGROUP1 here

File: source/MaterialXRuntime/RtLook.h [hunk 6]
```diff
@@ -1,19 +1,44 @@
...
 // All rights reserved.  See LICENSE.txt for license.
 //
 
 #ifndef MATERIALX_RTLOOK_H
 #define MATERIALX_RTLOOK_H
 
-/// @file
-/// TODO: Docs
+/// @file RtLook.h
+/// Specifictions for a set of classes related to material assignment.
```
Comment (niklasharrysson) [inline source/MaterialXRuntime/RtLook.h]: Typo: Specifictions

File: source/MaterialXTest/Runtime.cpp [hunk 7]
```diff
@@ -1076,4 +1078,42 @@ TEST_CASE("Runtime: Traversal", "[runtime]")
...
+
+    mx::RtPrim pa = stage->createPrim("matassign1", mx::RtMaterialAssign::typeName());
+    mx::RtMaterialAssign assign1(pa);
+    assign1.getCollection().addTarget(p1);
+
+    //TODO : Add rest of testing..
+    mx::FileSearchPath searchPath(mx::FilePath::getCurrentPath() / mx::FilePath("libraries"));
+    api->setSearchPath(searchPath);
+    mx::DocumentPtr doc = mx::createDocument();
+    loadLibraries({ "stdlib", "pbrlib" }, searchPath, doc);
```
Comment (niklasharrysson) [inline source/MaterialXTest/Runtime.cpp]: The preferred way to handle library loading is now through the `RtApi` class. You already have a handle to the api instance here, so you can use:

```
api->setSearchPath(searchPath);
api->loadLibrary(STDLIB);
api->loadLibrary(PBRLIB);
```

And then all nodedefs are automatically registered, so in order to create node instances from a nodedef you just give the nodedef name as the _typename_ to `createPrim`. For example if there is a nodedef for a `surfacematerial` node you can use:

const mx::RtToken surfaceMaterialDef("ND_surfacematerial");
mx::RtNode material1 = stage->createPrim("/material1", surfaceMaterialDef);

Comment (bernardkwok) [reply source/MaterialXTest/Runtime.cpp]: I've made the changes for this in the current code.

**Pull request merged**